scripts/install.sh

```diff
@@ -0,0 +1,156 @@
+  --values <file>             : Specify your own values file for the Helm chart.
+  --help                      : Show this help message and exit.
+
+Examples:
+  $0                               # Auto-detects router base and installs the Helm chart
+  $0 --router-base example.com     # Manually specifies the router base and installs the Helm chart
+  $0 --release-name myrelease      # Installs the Helm chart with the specified release name
+  $0 --generate-name               # Generates a name for the Helm release
+  $0 --values /path/to/values.yaml # Installs the Helm chart using the specified values file
+"
+}
+
+# Function to check if a command exists
+command_exists() {
+    command -v "$1" >/dev/null 2>&1
+}
+
+# Check for required commands
+if ! command_exists helm || ! command_exists oc; then
+    echo "Error: Required commands 'helm' and 'oc' are not available."
+    exit 1
+fi
+
+# Check if required files and directories exist
+HELM_CHART_DIR="$(dirname "$0")/../charts/backstage"
+DEFAULT_VALUES_FILE="$HELM_CHART_DIR/values.yaml"
+
+if [ ! -d "$HELM_CHART_DIR" ]; then
+    echo "Error: Helm chart directory not found at $HELM_CHART_DIR"
+    exit 1
+fi
+
+# Parse command-line arguments
+ROUTER_BASE=""
+RELEASE_NAME=""
+GENERATE_NAME=false
+NAMESPACE=""
+VALUES_FILE="$DEFAULT_VALUES_FILE"
+EXTRA_HELM_ARGS=""
+
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --router-base)
+            ROUTER_BASE="$2"
+            shift
+            ;;
+        --release-name)
+            RELEASE_NAME="$2"
+            shift
+            ;;
+        --generate-name)
+            GENERATE_NAME=true
+            ;;
+        --namespace)
+            NAMESPACE="$2"
+            shift
+            ;;
+        --values)
+            VALUES_FILE="$2"
+            shift
+            ;;
+        --help)
+            usage
+            exit 0
+            ;;
+        *)
+            EXTRA_HELM_ARGS+=" $1"
+            ;;
+    esac
+    shift
+done
+
+# Function to detect cluster router base
+detect_cluster_router_base() {
+    ROUTER_BASE=$(oc get ingress.config.openshift.io/cluster -o=jsonpath='{.spec.domain}')
+}
+
+# Detect cluster router base if not provided
+if [[ -z "$ROUTER_BASE" ]]; then
+    detect_cluster_router_base
+    if [[ -z "$ROUTER_BASE" ]]; then
+        echo "Error: Cluster router base could not be detected. Please provide it using the --router-base flag."
+        exit 1
+    fi
+fi
+
+# Detect namespace if not provided
+if [[ -z "$NAMESPACE" ]]; then
+    NAMESPACE=$(oc config view --minify --output 'jsonpath={..namespace}')
+    if [[ -z $NAMESPACE ]]; then
+        echo "Error: Namespace could not be detected. Please provide it using the --namespace flag."
+        exit 1
+    fi
+fi
+
+# Update Helm chart with the detected or provided router base if using default values file
+if [[ "$VALUES_FILE" == "$DEFAULT_VALUES_FILE" ]]; then
```

Comment:
You are right! since the `ROUTER_BASE` is always required and will either be provided by the user or auto-detected, there's no need to add the router base to the Helm arguments conditionally.